task/queue/queue.go

```diff
@@ -0,0 +1,80 @@
+package queue
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	"github.com/panjf2000/ants/v2"
+)
+
+// TaskQueue representation of task queue.
+type TaskQueue struct {
+	goPool             *ants.Pool
+	limit              int
+	numParallelRequest int
+
+	mutex sync.Mutex
+	queue *list.List
+}
+
+// New creates task queue which is processed by number of workers. Number of task is limited by limit.
+func New(numWorkers, limit int) (*TaskQueue, error) {
+	p, err := ants.NewPool(numWorkers, ants.WithPreAlloc(true), ants.WithNonblocking(true))
```

Comment:
Why do you want to limit the size of the pool? Shouldn't we reject device connections in case the system is becoming unresponsive rather than finding a constant value for the pool which might differ based on the HW where CoAP gateway is running?

Reply:
In ants you set maximum goroutines nad it is required parameter.
Coap-gw it will set by default to 2× logical cpu cores. You don't want to spawn lot's of goroutines.